### PR TITLE
Update Broadcast API to latest Refit standards for AB#14328

### PR DESCRIPTION
**Pull request by sslaws**
# Fixes or Implements [AB#14328](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/14328)

## Description
Updates the IBroadcastApi to latest Refit standards

## Testing

- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [X] Not Required

## UI Changes
Blazor Admin UI updated but no visible changes.

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/Admin/Client/Store/Broadcasts/BroadcastsEffects.cs [hunk 1]
```diff
@@ -61,17 +61,26 @@ public async Task HandleAddAction(BroadcastsActions.AddAction action, IDispatche
     {
         this.Logger.LogInformation("Adding broadcast");
 
-        ApiResponse<RequestResult<Broadcast>> response = await this.Api.Add(action.Broadcast).ConfigureAwait(true);
-        if (response.IsSuccessStatusCode && response.Content != null && response.Content.ResultStatus == ResultType.Success)
+        RequestResult<Broadcast> response = await this.Api.Add(action.Broadcast).ConfigureAwait(true);
+        try
```
Comment (MikeLyttle) [inline Apps/Admin/Client/Store/Broadcasts/BroadcastsEffects.cs:65]: The call to the API should be included inside the try block.

File: Apps/Admin/Client/Store/Broadcasts/BroadcastsEffects.cs [hunk 2]
```diff
@@ -84,17 +93,26 @@ public async Task HandleLoadAction(IDispatcher dispatcher)
     {
         this.Logger.LogInformation("Loading broadcasts");
 
-        ApiResponse<RequestResult<IEnumerable<Broadcast>>> response = await this.Api.GetAll().ConfigureAwait(true);
-        if (response.IsSuccessStatusCode && response.Content != null && response.Content.ResultStatus == ResultType.Success)
+        RequestResult<IEnumerable<Broadcast>> response = await this.Api.GetAll().ConfigureAwait(true);
+        try
```
Comment (MikeLyttle) [inline Apps/Admin/Client/Store/Broadcasts/BroadcastsEffects.cs:97]: The call to the API should be included inside the try block.

**Review by MikeLyttle — DISMISSED** [review]

File: Apps/Admin/Client/Store/Broadcasts/BroadcastsEffects.cs [hunk 3]
```diff
@@ -60,18 +60,26 @@ public BroadcastsEffects(ILogger<BroadcastsEffects> logger, IBroadcastsApi api)
...
+                this.Logger.LogInformation("Broadcast added successfully!");
+                dispatcher.Dispatch(new BroadcastsActions.AddSuccessAction(response));
+                return;
+            }
+
+            RequestError error = StoreUtility.FormatRequestError(null, response.ResultError);
+            this.Logger.LogError("Error adding broadcast, reason: {ErrorMessage}", error.Message);
+            dispatcher.Dispatch(new BroadcastsActions.AddFailAction(error));
+        }
+        catch (ApiException e)
```
[the remark this answers is not in the document]
Comment (MikeLyttle) [reply Apps/Admin/Client/Store/Broadcasts/BroadcastsEffects.cs:77]: Good catch.

[the remark this answers is not in the document]
Comment (sslaws) [reply Apps/Admin/Client/Store/Broadcasts/BroadcastsEffects.cs:77]: We never did handle HttpRequestException in this code

[the remark this answers is not in the document]
Comment (sslaws) [reply Apps/Admin/Client/Store/Broadcasts/BroadcastsEffects.cs:77]: Waiting for Brians update to merge into this PR.

**Review by MikeLyttle — APPROVED** [review]

**Review by rayking29 — APPROVED** [review]

**Pull request merged**